src/transactions/saga.ts

```diff
@@ -307,3 +313,58 @@ export function* transactionSaga() {
   yield* spawn(watchAddressToE164PhoneNumberUpdate)
   yield* spawn(watchPendingTransactions)
 }
+
+export function* handleTransactionReceiptReceived(
+  txId: string,
+  receipt: TransactionReceipt | CeloTxReceipt,
+  networkId: NetworkId
+) {
+  const tokensById = yield* select((state) => tokensByIdSelector(state, [networkId]))
+
+  // TODO: Receive feeCurrencyId from transaction request.
+  const feeCurrencyId = getTokenId(networkId)
+  const feeTokenInfo = tokensById[feeCurrencyId]
+
+  if (!feeTokenInfo) {
+    Logger.warn(`No information found for token ${feeCurrencyId} in network ${networkId}`)
+  }
+
+  const gasFeeInWei = new BigNumber(receipt.gasUsed.toString()).times(
+    new BigNumber(receipt.effectiveGasPrice.toString())
+  )
+
+  const baseDetails = {
+    transactionHash: receipt.transactionHash,
+    block: receipt.blockNumber.toString(),
+    status:
+      receipt.status === 'reverted' || !receipt.status
+        ? TransactionStatus.Failed
+        : TransactionStatus.Complete,
+  }
+
+  // Only add fee data in non-celo transactions
+  const feeDetails =
+    !feeTokenInfo || networkId === networkConfig.defaultNetworkId
+      ? {}
+      : {
+          fees: buildGasFees(feeTokenInfo, gasFeeInWei),
+        }
+
+  yield* put(
+    transactionConfirmed(txId, {
+      ...baseDetails,
+      ...feeDetails,
+    })
+  )
+}
+
+const buildGasFees = (feeCurrencyInfo: BaseToken, gasFeeInWei: BigNumber): Fee[] => [
+  //
+  {
+    type: 'SECURITY_FEE',
+    amount: {
+      value: gasFeeInWei.shiftedBy(-feeCurrencyInfo.decimals).toFixed(),
+      tokenId: feeCurrencyInfo.tokenId,
+    },
+  },
+]
```

Comment:
nit: using the `function` syntax is easier to spot functions while reading the code IMHO.
```suggestion
function buildGasFees(feeCurrencyInfo: BaseToken, gasFeeInWei: BigNumber): Fee[] {
  return [
    //
    {
      type: 'SECURITY_FEE',
      amount: {
        value: gasFeeInWei.shiftedBy(-feeCurrencyInfo.decimals).toFixed(),
        tokenId: feeCurrencyInfo.tokenId,
      },
    },
  ]
}
```